plans/internal/planproto/planfile.proto

```diff
@@ -105,21 +105,26 @@ message ResourceInstanceChange {
     // set, the change applies to the instance's current object.
     string deposed_key = 7;
 
+    // provider is the address of the provider configuration that this change
+    // was planned with, and thus the configuration that must be used to
+    // apply it.
+    string provider = 8;
+
     // Description of the proposed change. May use "create", "read", "update",
     // "replace" and "delete" actions. "no-op" changes are not currently used here
     // but consumers must accept and discard them to allow for future expansion.
-    Change change = 8;
+    Change change = 9;
```

Comment:
If we'd already shipped this then we would of course not update the already-assigned indexes to retain compatibility, but since this is still in development I thought it nicer to keep the indices increasing monotonically for now.